addons/BlocklyCodeEditor/test/commandsTests.js

```diff
@@ -1,31 +1,29 @@
-var setUpFunction = function () {
-    this.presenter = AddonBlocklyCodeEditor_create();
-    this.blocklyStateValue = JSON.stringify({"code":"<xml xmlns=\"http://www.w3.org/1999/xhtml\"><block type=\"text_prompt_ext\" id=\"mFW}==osP6Y*]az/49e-\" x=\"51\" y=\"153\"><mutation type=\"TEXT\"></mutation><field name=\"TYPE\">TEXT</field></block><block type=\"text_prompt_ext\" id=\")Lj~:VnU]2*[`|PbMfQ%\" x=\"131\" y=\"219\"><mutation type=\"TEXT\"></mutation><field name=\"TYPE\">TEXT</field><value name=\"TEXT\"><block type=\"math_single\" id=\"I@js!Bu?k7i1KWD5?^:T\"><field name=\"OP\">ROOT</field><value name=\"NUM\"><block type=\"math_number\" id=\"d$sG;0/0;*22[R!@Y`+%\"><field name=\"NUM\">0</field></block></value></block></value></block><block type=\"variables_set\" id=\".HdMVm$9ht$d|RcviYIG\" x=\"144\" y=\"268\"><field name=\"VAR\">item</field></block></xml>","isVisible":true});
-    this.presenter.configuration.workspace = Blockly.inject("body", {
-        toolbox: "<xml></xml>",
-        sounds: false,
-        maxBlocks: 0
-    });
+TestCase('[Blockly_Code_Editor] workspace code tests', {
+    setUp: function () {
+        this.presenter = AddonBlocklyCodeEditor_create();
+        this.presenter.configuration.maxBlocks = 10;
+        this.presenter.configuration.workspace = {
+            'toolbox': '<xml></xml>',
+            'sounds': false,
+            'maxBlocks': this.presenter.configuration.maxBlocks
+        };
 
-    var self = this;
-    this.stubs = {
-        setVisibility: sinon.stub(this.presenter, 'setVisibility', function (isVisible) {
-            self.presenter.configuration.isVisible = isVisible;
-        })
-    };
-};
+        Blockly.JavaScript = {
+            workspaceToCode: sinon.mock()
+        };
+        Blockly.JavaScript.workspaceToCode.returns('<xml sounds="false" maxBlocks="10"></xml>');
```

Comment:
Jeśli już musisz nadpisać bibliotekę w ten sposób, to zapisz gdzieś dotychczasową wartość "na boku" i przywróć ją po zakończeniu testów.